addons/mail/static/src/models/thread/thread.js

```diff
@@ -416,6 +416,23 @@ function factory(dependencies) {
             this.update({ areAttachmentsLoaded: true });
         }
 
+        async fetchUpdateSuggestedPartners() {
+            const result = await this.async(() => this.env.services.rpc({
+                route: '/mail/get_suggested_recipients',
+                params: {
+                    model: this.model,
+                    res_ids: [this.id],
+                },
+            }));
+            if (result[this.id]) {
+                this.update({
+                    suggestedPartners: [['insert-and-replace', result[this.id].map(data => {
+                        return this.env.models['mail.suggested_partner'].convertData(data);
+                    })]],
+                });
+            }
```

Comment:
```js
const { [this.id]: data } = await ...
if (!data) {
   return;
}
this.update({ ... });
```

Reply:
unresolved comment:

```js
const { [this.id]: data } = await ...
if (!data) {
   return;
}
this.update({ ... });
```

Reply:
I don't really like this pattern, feel over complicated. On the first line you already use 2 "Javascript magic". It can be usefull sometimes, but I don't think it's the case here.
I don't really understand how it's better. If it's for the `return;` an `else` whould have done it. If it's to avoid `result[this.id]`, I don't understand the issue. 

I have make the change anyway.

Reply:
Code is hard to read because of `result[this.id]`. This being a variable is better (equivalent to `data` in suggested code). Maybe this looks like "Javascript magic" for you because you are not used to object destructuring + renaming + dynamic prop name evaluation, which are all well-defined ES6 features (none of them are hacks). Using these features of JS makes code short and easy to understand.

The preemptive `return;` is best because too many indent levels reduce readability of code.

Reply:
> Code is hard to read because of result[this.id].

I disagree, but let's play with that statement. You just moved the `this.id` inside the constant declaration. It make the `data` var magic since it come from nowhere and the constant declaration unreadable (3 operations on one line).

> Using these features of JS makes code short and easy to understand.

Short code != easy to understand. Most of the time, it's the opposite...
You really hurt my love for the KISS principle, I'd rather do:
```javascript
const result = await...
const data = result[this.id];
```
Then using ES6 features.

Reply:
> > Code is hard to read because of result[this.id].
> 
> I disagree, but let's play with that statement. You just moved the `this.id` inside the constant declaration. It make the `data` var magic since it come from nowhere and the constant declaration unreadable (3 operations on one line).

`data` doesn't come from nowhere, `const X = Y;`,`X` contains declaration of variables. Sure it needs some getting used to this syntax, but I really don't think this is unreadable. Also I've put this in a single line, but it could also be in multi-line like this:

```js
const {
   [this.id]: data,
} = await ...
```

> > Using these features of JS makes code short and easy to understand.
> 
> Short code != easy to understand. Most of the time, it's the opposite...
> You really hurt my love for the KISS principle

Fully agree short code != easy to understand, would even go further and say that short code != good code.

The KISS principle is usually good, but not applicable sometimes ("simple" not always "short"), so... even the things we love have their flaws... Not saying you should dislike this principle, quite the contrary, but don't just blindly follow this principle to the letter...

> I'd rather do:
> 
> ```js
> const result = await...
> const data = result[this.id];
> ```
> 
> Then using ES6 features.

I don't mind doing this in 2 steps like this, the only "slight" drawback is that you need to name the value of `await` ("`result`") and you need to use it just to access an entry in it. `array[expression]` quickly makes things hard to read, which is why I tend to avoid writing something like this.

Reply:
> I don't mind doing this in 2 steps like this, the only "slight" drawback is that you need to name the value of `await` ("`result`") and you need to use it just to access an entry in it. `array[expression]` quickly makes things hard to read, which is why I tend to avoid writing something like this.

...not that it's bad to have `result` variable... but IMO it puts too much importance on `result` than it should.

To give an analogy: it's like when you listen to a music performance and the notes and rhythm is perfect but the voicing is not...